File: longestCommonPrefix/longestCommonPrefix.py
def longestCommonPrefix(v: list[str]) -> str:
    
    ans = ""
    v = sorted(v)
    
    f = v[0]
    l = v[-1]

    for i in range(min(len(f), len(l))):
        if f[i] == l[i]:
            ans += f[i]
        else:
            break
    
    return ans


if __name__ == "__main__":
    v = ["flower","flow","flight"]
    ans = longestCommonPrefix(v)
    print(ans)
    
    

File: longestCommonPrefix/longestCommonPrefix.cpp
#include<iostream>
#include<vector>
using namespace std;

string longestCommonPrefix(vector<string> &strs){
    string ans = "";
    sort(strs.begin(), strs.end());
        
    int num = strs.size();

    string f = strs[0];
    string l = strs[num-1];

    for (int i=0; i<min(f.size(), l.size()); i++){
        if (f[i] != l[i]) return ans;
        ans += f[i];
    }
    return ans;
}

int main(){
    vector<string> strs = {"flower","flow","flight"};
    cout << longestCommonPrefix(strs) << endl;
    return 0;
}